src/main/java/org/linksharing/server/auth/AuthController.java

```diff
@@ -27,23 +29,23 @@ public String showLoginPage() {
     }
 
     @GetMapping("/register")
-    public String showRegistrationPage() {
+    public String showRegistrationPage(User user) {
         return "registration_page.html";
     }
 
     @PostMapping("/register")
-    public String registerUser(@RequestParam("username") String username,
-                               @RequestParam("email") String email,
-                               @RequestParam("password") String password) {
+    public String registerUser(@Valid User user, BindingResult result) {
 
-        if (userRepository.existsByEmail(email)) {
-            return "redirect:/login?userExists";
+        if (result.hasErrors()) {
+            return "registration_page.html";
         }
 
-        User user = new User();
-        user.setUsername(username);
-        user.setEmail(email);
-        user.setPassword(passwordEncoder.encode(password));
+        String pass = user.getPassword();
+        user.setPassword(passwordEncoder.encode(pass));
+
+        if (userRepository.existsByEmail(user.getEmail())) {  //doesn't work for some reason
```

Comment:
Looks like we need to add this to SecurityConfig in order for it to work:
```java
        http.authorizeHttpRequests(c -> c
               ...
                .requestMatchers(antMatcher("/login"))
                .permitAll()

```

Reply:
You might want to add a message for users if they redirected to "/login?userExists"...

Reply:
> ```java
> .requestMatchers(antMatcher("/login"))
>                 .permitAll()
> ```

That's absolutely right. How do you.....?  Whatever)))

Reply:
> You might want to add a message for users if they redirected to "/login?userExists"...

Am I right here? 
"/login?userExists"   userExists is a URL parameter. And thymeleaf should see this parameter somehow and show the special custom div that I will define in login_page.html ?

Reply:
Yes, this should work

Reply:
Done